tests/Integration/Console/ConsoleApplicationTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Illuminate\Tests\Integration\Console;
+
+use Illuminate\Console\Command;
+use Orchestra\Testbench\TestCase;
+use Illuminate\Contracts\Console\Kernel;
+
+class ConsoleApplicationTest extends TestCase
+{
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->app[Kernel::class]->registerCommand(new FooCommandStub);
+    }
+
+    public function test_artisan_call_using_command_name()
+    {
+        $exitCode = $this->artisan('foo:bar', [
+            'id' => 1,
+        ]);
+
+        $this->assertEquals($exitCode, 0);
+    }
+
+    public function test_artisan_call_using_command_class()
+    {
+        $exitCode = $this->artisan(FooCommandStub::class, [
+            'id' => 1,
+        ]);
+
+        $this->assertEquals($exitCode, 0);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Console\Exception\CommandNotFoundException
+     */
+    public function test_artisan_call_invalid_command_name()
+    {
+        $this->artisan('foo:bars');
```

Comment:
@KennedyTedesco this test freezes entirely on master branch. Can you fix it?

Reply:
@taylorotwell We can fix on our side by evaluating if the command really exists. What you think about it?

https://github.com/KennedyTedesco/framework/commit/bc20aaa0ef07682cf0ef7d4c9eb1b11456c0f07e

I can send a pull request if it's fine.

Reply:
OK